src/main/java/net/pms/network/SpeedStats.java

```diff
@@ -150,17 +150,24 @@ private Integer doCall() throws Exception {
 				}
 			}
 			double speedInMbits1 = bps / (cnt * 1000000);
-			LOGGER.info("Renderer {} has an estimated network speed of {} Mb/s", rendererName, speedInMbits1);
-			int speedInMbits = (int) speedInMbits1;
-			if (speedInMbits1 < 1.0) {
+			int speedInMbits;
+			if (speedInMbits1 == Double.NaN) {
```

Comment:
Instead of catching it here when the `NaN` has occurred, which means that `doPing` has likely run 5 times, it might be better to move the check higher up.

Looking at https://github.com/UniversalMediaServer/UniversalMediaServer/blob/master/src/main/java/net/pms/network/SpeedStats.java#L199 maybe you could remove that `try...catch` and instead allow that method to throw the error, and then we catch it here?

What do you think?